### PR TITLE
Cmake: fix 'make install' and use dynamic linking by default for UNIX

### PR DESCRIPTION
**Pull request by weinrank**

### CODE REVIEW

File: usrsctplib/CMakeLists.txt [hunk 1]
```diff
@@ -275,7 +275,9 @@ ENDIF ()
 
 INSTALL (TARGETS usrsctp usrsctp-static DESTINATION ${CMAKE_INSTALL_LIBDIR})
 #INSTALL (DIRECTORY include DESTINATION ${CMAKE_INSTALL_PREFIX})
-INSTALL (FILES ${usrsctp_root_HEADERS} DESTINATION include/usrsctp)
-INSTALL (FILES ${usrsctp_netinet_HEADERS} DESTINATION include/usrsctp/netinet)
-INSTALL (FILES ${usrsctp_netinet6_HEADERS} DESTINATION include/usrsctp/netinet6)
-INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/usrsctp.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+#INSTALL (FILES ${usrsctp_root_HEADERS} DESTINATION include/usrsctp)
+#INSTALL (FILES ${usrsctp_netinet_HEADERS} DESTINATION include/usrsctp/netinet)
```
Comment (tuexen) [inline usrsctplib/CMakeLists.txt]: Why is this line needed? What are you installing there?

File: usrsctplib/CMakeLists.txt [hunk 2]
```diff
@@ -275,7 +275,9 @@ ENDIF ()
...
+#INSTALL (FILES ${usrsctp_netinet6_HEADERS} DESTINATION include/usrsctp/netinet6)
```
Comment (tuexen) [inline usrsctplib/CMakeLists.txt]: Why is this line needed? What are you installing there?

File: usrsctplib/CMakeLists.txt [hunk 3]
```diff
@@ -275,7 +275,9 @@ ENDIF ()
 
 INSTALL (TARGETS usrsctp usrsctp-static DESTINATION ${CMAKE_INSTALL_LIBDIR})
 #INSTALL (DIRECTORY include DESTINATION ${CMAKE_INSTALL_PREFIX})
-INSTALL (FILES ${usrsctp_root_HEADERS} DESTINATION include/usrsctp)
-INSTALL (FILES ${usrsctp_netinet_HEADERS} DESTINATION include/usrsctp/netinet)
-INSTALL (FILES ${usrsctp_netinet6_HEADERS} DESTINATION include/usrsctp/netinet6)
-INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/usrsctp.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+#INSTALL (FILES ${usrsctp_root_HEADERS} DESTINATION include/usrsctp)
```
Comment (tuexen) [inline usrsctplib/CMakeLists.txt]: Why do you want to create a directory to install a single file? Is that the normal way? Just wondering...

Comment (weinrank) [inline usrsctplib/CMakeLists.txt]: I don't - that's why i commented it out. 
This line is from the initial CMAKE commit

File: usrsctplib/CMakeLists.txt (hunk 1 above)
Comment (weinrank) [inline usrsctplib/CMakeLists.txt]: Currently all header-files from netinet are copied to this directory.
I don't think this is necessary...

File: usrsctplib/CMakeLists.txt (hunk 2 above)
Comment (weinrank) [inline usrsctplib/CMakeLists.txt]: Like in the line above the (single) header file from netinet6 is copied to this directory.

File: usrsctplib/CMakeLists.txt (hunk 3 above)
Comment (tuexen) [inline usrsctplib/CMakeLists.txt]: Missed that it is commented out. Why not remove it?

File: usrsctplib/CMakeLists.txt (hunk 1 above)
Comment (tuexen) [inline usrsctplib/CMakeLists.txt]: Don't install any header file except usrsctp.h. Only that is the interface.

File: usrsctplib/CMakeLists.txt (hunk 2 above)
Comment (tuexen) [inline usrsctplib/CMakeLists.txt]: See above...

File: usrsctplib/CMakeLists.txt (hunk 1 above)
Comment (weinrank) [inline usrsctplib/CMakeLists.txt]: This is exactly what we achieve by this patch.
The behavior before this patch is to copy all files to <includedir>/usrsctp/

With this patch *only* the usrsctp.h is copied  -  and not to <includedir>/usrsctp/ but to <includedir>/ - this is how the traditional method does.

File: usrsctplib/CMakeLists.txt (hunk 3 above)
Comment (weinrank) [inline usrsctplib/CMakeLists.txt]: Experimental technology - so it's easier to track changes to the initial version.

Comment (tuexen) [inline usrsctplib/CMakeLists.txt]: Since we don't want to do that take it out.
If you want to have some historical stuff. Commit it as you got it and then take the lines out...

**Comment by weinrank** [comment]
@tuexen Cleaned the cmake file and removed old .cvsignore
